### ISSUE TITLE
`Rc<DynamicIndex>` cannot be used `with_new_thread`

### ISSUE THREAD

**Issue by rcastill** [AaronErhardt/relm4#58]
I was trying to setup somethig like the following:

```
AppModel (App)
|-- DataModel (Component)
```

`AppMsg` (`AppModel::Msg`) contains a `Weak<DynamicIndex>` in one of the possible messages because I have a `FactoryVecDeque`. That means I cannot `relm4::RelmComponent::with_new_thread(...)` for `DataModel` since `ParentModel::Msg: !Send`.

My workaround for that was to create an intermediate component `InterModel` such that `InterModel::Msg: Send` because it does not need `Weak / Rc` fields. Like this:

```
AppModel (App)
|-- InterModel (Component)
|---- DataModel (Component)
```

The only drawback that I find is that I have to "repeat" a message to comunicate from `AppModel -> DataModel` and viceversa (through `InterModel`).

Is this approach OK? or is there a more idiomatic way to achieve this?

PS: I looked into Message Handlers, but it seemed that I would have had to spawn a new thread, that also required `AppModel::Msg: Send`.

**Comment by AaronErhardt** [comment]
That's definitely a problem. It's not great that you can't send messages to the parent if it uses a `FactoryVecDeque`. I will have a look at this, but Rust could make it a bit hard to find a solution for this. Maybe I have to use `Arc<Mutex<usize>>` as Index eventually, even it it would be unnecessary overhead.

Regarding `relm4::RelmComponent::with_new_thread(...)`, this method will likely be removed in the upcoming stable release. As [explained in the book](https://aaronerhardt.github.io/relm4-book/book/threads_and_async.html#running-a-component-on-a-different-thread) it always needed a weird hack to work and became hard to maintain. I think using workers or message handlers is more suitable for CPU-heavy operations in the update method.

**Comment by AaronErhardt** [comment]
I've published version 0.4.0-beta.3 that's not affected by this issue anymore. `DynamicIndex` doesn't need an external `Rc` anymore and uses fragile for the `Send` implementation. This means it's still not allowed to send `DynamicIndex` to other threads but at least the type implements `Send` and only panics when used on other threads. Therefore the message type can now also implement `Send`. Since I don't think there's a real use case for sending the dynamic index to other threads I think this issue should be solved.

**Comment by rcastill** [comment]
That's great @AaronErhardt! 

Thanks
